src/bitmessagemain.py

```diff
@@ -73,17 +77,17 @@ def connectToStream(streamNumber):
     selfInitiatedConnections[streamNumber] = {}
 
     if isOurOperatingSystemLimitedToHavingVeryFewHalfOpenConnections():
-        # Some XP and Vista systems can only have 10 outgoing connections at a time.
+        # Some XP and Vista systems can only have 10 outgoing connections
+        # at a time.
         state.maximumNumberOfHalfOpenConnections = 9
     else:
         state.maximumNumberOfHalfOpenConnections = 64
-    try:
-        # don't overload Tor
-        if BMConfigParser().get('bitmessagesettings', 'socksproxytype') != 'none':
-            state.maximumNumberOfHalfOpenConnections = 4
-    except:
-        pass
-    
+
+    # don't overload Tor
+    if BMConfigParser().safeGet(
```

Comment:
I'm not fully sure the safeGet functions handle missing values like this. Perhaps instead of this change, just add a default socksproxytype in bmconfigparser and remove 'none'?

Reply:
It does. And it's already used in many places in code doing the same `try: ... except:`

Look at the [def](/Bitmessage/PyBitmessage/blob/master/src/bmconfigparser.py#L72)

Reply:
I know it's used in other places but that doesn't mean it works correctly. Maybe it does. If you tested it then it's fine.